apps/root/src/hooks/useTransactionsHistory.tsx

```diff
@@ -418,50 +398,33 @@ function useTransactionsHistory(): {
           },
         } as TransactionEvent;
       });
-      const resolvedEvents = await Promise.all(eventsPromises);
-      return compact(resolvedEvents);
-    },
-    []
-  );
 
-  const transformEvents = React.useCallback(
-    async (events: TransactionApiEvent[], tokenList: TokenListByChainId, userWallets: string[]) => {
-      if (!events) return [];
-      const eventsPromises = parseMultipleTransactionApiEventsToTransactionEvents(
-        events,
-        dispatch,
+      return compact(eventsPromises);
+    };
+
+    if (!isLoadingTokenLists && historyEvents && !isLoading) {
+      // if (!isLoadingTokenLists && historyEvents && !isLoading && (!isEqual(prevPendingTransactions, pendingTransactions) || !isEqual(prevHistoryEvents, historyEvents))) {
+      if (!historyEvents) return;
+      const resolvedEvents = parseMultipleTransactionApiEventsToTransactionEvents(
+        historyEvents,
         tokenList,
-        userWallets
-      ).map((promise) =>
-        promise.catch((e) => {
-          console.error('Failed to parse event', e);
-          return null;
-        })
+        storedWallets.map(({ address }) => address)
       );
 
-      const resolvedEvents = compact(await Promise.all(eventsPromises));
-
-      const pendingEvents = await transformPendingEvents(pendingTransactions, tokenList, userWallets);
+      const pendingEvents = transformPendingEvents(
+        pendingTransactions,
+        storedWallets.map(({ address }) => address)
+      );
 
       resolvedEvents.unshift(...pendingEvents);
 
       setParsedEvents(resolvedEvents);
-    },
-    [pendingTransactions]
-  );
-
-  React.useEffect(() => {
-    if (!isLoadingTokenLists && historyEvents && !isLoading) {
-      void transformEvents(
-        historyEvents,
-        tokenListByChainId,
-        storedWallets.map(({ address }) => address)
-      );
 
       if (indexing) dispatch(cleanTransactions({ indexing }));
     }
     // Whenever the events, the token list or any pending transaction changes, we want to retrigger this
-  }, [historyEvents, indexing, isLoadingTokenLists, isLoading, hasPendingTransactions]);
+  }, [historyEvents, indexing, isLoadingTokenLists, isLoading, tokenList, pendingTransactions]);
+  // }, [historyEvents, indexing, isLoadingTokenLists, isLoading, tokenList, pendingTransactions, prevHistoryEvents, prevPendingTransactions]);
```

Comment:
There are some commented lines we should remove, here and other ones related with the usePrevious hook